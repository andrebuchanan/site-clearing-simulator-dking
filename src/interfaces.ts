export interface IBulldozerPosition {
  xPos: number;
  yPos: number;
}

export interface ISiteMapProps {
  siteMap: string[][],
  bulldozerPosition: IBulldozerPosition
}

export enum EBulldozerDirection {
  north,
  east,
  south,
  west
}

export enum EUserCommand {
  advance,
  left,
  right,
  quit
}

export interface IUserCommand {
  command: EUserCommand;
  value: number;
}

export interface ISquareProps {
  landType: string
}

export interface IReduxAction {
  type: string;
  payload: {}
}

export interface IMapBorders {
  eastBorder: number,
  southBorder: number
}