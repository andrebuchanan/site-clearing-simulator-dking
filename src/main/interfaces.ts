export interface IBulldozerPosition {
  xPos: number;
  yPos: number;
}

export interface ISiteMapProps {
  siteMap: string[][],
  bulldozerPosition: IBulldozerPosition,
  UpdateSiteMapCallback(newSiteMap: string[][]): void
}

export enum EBulldozerDirection {
  north,
  east,
  south,
  west
}

export enum EUserCommand {
  advance,
  left,
  right,
  quit
}

export interface IUserCommand {
  command: EUserCommand;
  value: number;
}

export interface ISquareProps {
  landType: string
}