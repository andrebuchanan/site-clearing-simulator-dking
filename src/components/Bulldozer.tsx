import React, { useState, useEffect } from 'react';
import eastImg from "../images/bulldozerEast.png";
import westImg from "../images/bulldozerWest.png";
import southImg from "../images/bulldozerSouth.png";
import northImg from "../images/bulldozerNorth.png";
import { EBulldozerDirection, IBulldozerProps } from "../interfaces";
import { connect } from 'react-redux';

const mapStateToProps = (state:  any/*TODO */) => {
  return {
    bulldozerDirection: state.bulldozerDirection
  };
};


const ConnectedBulldozer = ({bulldozerDirection}: IBulldozerProps) => {

  const [ currentImage, setCurrentImage] = useState<string>();

  const determineBulldozerOrientation = () => {
    switch(bulldozerDirection){
      case EBulldozerDirection.east:
        setCurrentImage(eastImg);
        break;
      case EBulldozerDirection.west:
        setCurrentImage(westImg);
        break;
      case EBulldozerDirection.south:
        setCurrentImage(southImg);
        break;
      case EBulldozerDirection.north:
        setCurrentImage(northImg);
        break;
    }
  }

  useEffect(() => {
    determineBulldozerOrientation();
  }, [bulldozerDirection])

  return(
    <img className="bulldozer-img" src={currentImage}></img>
  );
}

const Bulldozer = connect(mapStateToProps)(ConnectedBulldozer)

export default Bulldozer;