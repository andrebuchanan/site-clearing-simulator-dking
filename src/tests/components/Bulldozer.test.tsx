import configureStore from 'redux-mock-store';
import { EBulldozerDirection } from '../../interfaces';
import React, { JSXElementConstructor } from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Bulldozer from '../../components/Bulldozer';
 
const mockStore = configureStore([]);

describe("Connected Redux Bulldozer Component", () => {

    let store;
    let BulldozerComponent: any;
    beforeEach(() => {
        store = mockStore({
            bulldozerDirection: EBulldozerDirection.east,
        });

        BulldozerComponent = renderer.create(
            <Provider store={store}>
                <Bulldozer />
            </Provider>
        );
    });

    it("should render with given state from Redux store", () => {
        expect(BulldozerComponent.toJSON()).toMatchSnapshot();
    })
});