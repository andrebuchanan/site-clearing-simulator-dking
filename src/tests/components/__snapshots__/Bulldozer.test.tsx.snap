// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Redux Bulldozer Component should render Bulldozer Component 1`] = `
<img
  className="bulldozer-img"
/>
`;

exports[`Connected Redux Bulldozer Component should render with given state from Redux store 1`] = `
<img
  className="bulldozer-img"
/>
`;
